File: ttshare_mobile/lib/services/cookie_manager.dart
import 'dart:convert';
import 'package:shared_preferences/shared_preferences.dart';
import '../models/cookie_entry.dart';

class CookieManager {
  static const _keyPrefix = 'cookies_';
  static const _domainsKey = 'cookie_domains';

  Future<List<CookieEntry>> getAllCookies() async {
    final prefs = await SharedPreferences.getInstance();
    final domains = prefs.getStringList(_domainsKey) ?? [];
    final cookies = <CookieEntry>[];
    for (final domain in domains) {
      final jsonStr = prefs.getString('$_keyPrefix$domain');
      if (jsonStr != null) {
        cookies.add(CookieEntry.fromJson(
            json.decode(jsonStr) as Map<String, dynamic>));
      }
    }
    return cookies;
  }

  Future<CookieEntry?> getCookies(String domain) async {
    final prefs = await SharedPreferences.getInstance();
    final jsonStr = prefs.getString('$_keyPrefix$domain');
    if (jsonStr == null) return null;
    return CookieEntry.fromJson(json.decode(jsonStr) as Map<String, dynamic>);
  }

  Future<void> saveCookies(String domain, String cookieData) async {
    final prefs = await SharedPreferences.getInstance();
    final now = DateTime.now();
    final existing = await getCookies(domain);
    final entry = CookieEntry(
      domain: domain,
      cookieData: cookieData,
      createdAt: existing?.createdAt ?? now,
      updatedAt: now,
    );
    await prefs.setString('$_keyPrefix$domain', json.encode(entry.toJson()));
    final domains = prefs.getStringList(_domainsKey) ?? [];
    if (!domains.contains(domain)) {
      domains.add(domain);
      await prefs.setStringList(_domainsKey, domains);
    }
  }

  Future<void> clearDomain(String domain) async {
    final prefs = await SharedPreferences.getInstance();
    await prefs.remove('$_keyPrefix$domain');
    final domains = prefs.getStringList(_domainsKey) ?? [];
    domains.remove(domain);
    await prefs.setStringList(_domainsKey, domains);
  }

  Future<void> clearAll() async {
    final prefs = await SharedPreferences.getInstance();
    final domains = prefs.getStringList(_domainsKey) ?? [];
    for (final domain in domains) {
      await prefs.remove('$_keyPrefix$domain');
    }
    await prefs.remove(_domainsKey);
  }
}

File: ttshare_android/app/src/main/java/com/ttshare/service/CookieManager.kt
package com.ttshare.service

import android.content.SharedPreferences
import android.content.Context
import java.io.File

class CookieManager(private val prefs: SharedPreferences) {

    data class CookieEntry(
        val domain: String,
        val cookieData: String
    )

    fun getCookies(domain: String): CookieEntry? {
        val data = prefs.getString("cookie_$domain", null) ?: return null
        return CookieEntry(domain, data)
    }

    fun saveCookies(domain: String, cookieData: String) {
        prefs.edit().putString("cookie_$domain", cookieData).apply()
        val domains = prefs.getStringSet("cookie_domains", mutableSetOf())?.toMutableSet()
            ?: mutableSetOf()
        domains.add(domain)
        prefs.edit().putStringSet("cookie_domains", domains).apply()
    }

    fun getAllDomains(): List<String> {
        return prefs.getStringSet("cookie_domains", emptySet())?.toList() ?: emptyList()
    }

    fun getAllCookies(): List<CookieEntry> {
        return getAllDomains().mapNotNull { domain ->
            getCookies(domain)?.let { CookieEntry(domain, it.cookieData) }
        }
    }

    fun clearDomain(domain: String) {
        prefs.edit().remove("cookie_$domain").apply()
        val domains = prefs.getStringSet("cookie_domains", mutableSetOf())?.toMutableSet()
            ?: mutableSetOf()
        domains.remove(domain)
        prefs.edit().putStringSet("cookie_domains", domains).apply()
    }

    fun clearAll() {
        val domains = getAllDomains()
        val editor = prefs.edit()
        for (domain in domains) {
            editor.remove("cookie_$domain")
        }
        editor.remove("cookie_domains").apply()
    }
}
